test/test-network/systemd-networkd-tests.py.in

```diff
@@ -2394,6 +2395,24 @@ def test_dhcp_client_route_remove_on_renew(self):
     if ubsan_options:
         env.update({ 'UBSAN_OPTIONS' : ubsan_options })
 
+    if ns.build_dir:
+        with open (os.path.join(ns.build_dir, 'systemd-runtest.env')) as f:
+            for line in f.read().splitlines():
+                if 'SYSTEMD_TEST_DATA' in line:
+                    conf_dir = os.path.join(line.split('=')[1], 'test-network/conf')
+    else:
+        with open (os.path.join(os.path.dirname(os.path.abspath(__file__)), '../../systemd-runtest.env')) as f:
+            for line in f.read().splitlines():
+                if 'SYSTEMD_TEST_DATA' in line:
+                    conf_dir = os.path.join(line.split('=')[1], 'test-network/conf')
```

Comment:
You can drop `with` here, it makes no difference. Also use normal indexing to avoid explicit check:
```python
if ns.build_dir:
    dirpath = ns.build_dir,
else
    dirpath = os.path.dirname(os.path.abspath(__file__)), '../..'
f = open(os.path.join(*dirpath, systemd-runtest.env'))
for line in f:
     if 'SYSTEMD_TEST_DATA' in line:
          conf_dir = os.path.join(line.split('=')[1], 'test-network/conf')
else:
     conf_dir = os.path.join(os.environ['SYSTEMD_TEST_DATA'], 'test-network')
```